### ISSUE TITLE
Fails with Javascript "in operator" error

### ISSUE THREAD

**Issue by waldoj** [veltman/address-book#1]
I'm trying to geocode this file:

``` csv
street,city,zip,latitude,longitude
123 E. Main St.,Charlottesville,22902,,
```

With this command:

```
$ csvgeocode --url "http://geoservices.tamu.edu/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsed_V04_01.aspx?apiKey=abcdefghijklmnopqrstuvwxyz&version=4.01&streetAddress={{street}}&city={{city}}&state=VA" test.csv
```

It's failing, with this error:

```
/usr/local/lib/node_modules/csvgeocode/src/csvgeocode.js:161
    } else if ("lat" in result && "lng" in result) {
                        ^
TypeError: Cannot use 'in' operator to search for 'lat' in undefined
    at handleResponse (/usr/local/lib/node_modules/csvgeocode/src/csvgeocode.js:161:25)
    at Request._callback (/usr/local/lib/node_modules/csvgeocode/src/csvgeocode.js:133:9)
    at Request.self.callback (/usr/local/lib/node_modules/csvgeocode/node_modules/request/request.js:344:22)
    at Request.emit (events.js:110:17)
    at Request.<anonymous> (/usr/local/lib/node_modules/csvgeocode/node_modules/request/request.js:1239:14)
    at Request.emit (events.js:129:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/csvgeocode/node_modules/request/request.js:1187:12)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
    at process._tickCallback (node.js:355:11)
```

I'd figured that `result` isn't defined because the file can't be opened, but running in verbose mode precedes the above error with this:

```
123 E. Main St.,Charlottesville,22902,
Parsing error: SyntaxError: Unexpected number
```

I don't know what's up with the SyntaxError, but the point is that it _is_ retrieving the first data line from the file.

I'm running Node v0.12.0.

Any ideas?

**Comment by veltman** [comment]
Yup, you need to at least supply `--handler tamu` as an option if that's the geocoder you're using.

**Comment by waldoj** [comment]
I'm so sorry, I neglected to include that in my example. But even with that option (and without it), the problem remains.

**Comment by veltman** [comment]
Hmm, the following command works fine for me:

```
csvgeocode --handler tamu --url "http://geoservices.tamu.edu/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsed_V04_01.aspx?apiKey=$TAMU_API_KEY&version=4.01&streetAddress={{street}}&city={{city}}&state=VA" test.csv geocoded.csv
```

Are you sure you're using a valid API key?  What happens if you do this request directly (subbing in your key)?

```
curl -- "http://geoservices.tamu.edu/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsed_V04_01.aspx?apiKey=WALDOS_API_KEY&version=4.01&streetAddress=123+E+Main+St&city=Charlottesville&state=VA"
```

**Comment by veltman** [comment]
Also, in the course of investigating this, I found an unrelated bug, yay! So I'd also update to the latest version if you can.

**Comment by veltman** [comment]
When I run

```
$ TAMU_API_KEY=whatever
$ csvgeocode --url "http://geoservices.tamu.edu/Services/Geocode/WebService/GeocoderWebServiceHttpNonParsed_V04_01.aspx?apiKey=$TAMU_API_KEY&version=4.01&streetAddress={{street}}&city={{city}}&state=VA" --handler tamu test.csv
```

Against your CSV I get:

```
street,city,zip,latitude,longitude
123 E. Main St.,Charlottesville,22902,38.0307154056089,-78.4816126911609
```

**Comment by waldoj** [comment]
I've got it. Wow, was that ever non-obvious. It turns out that I had omitted a single character from my API key, presumably a copy-and-paste error when I grabbed it off TAMU's website initially. It [looks like](https://geoservices.tamu.edu/Services/Geocode/WebService/) TAMU's API doesn't have a mechanism of reporting that an API key is incorrect—it just fires back with a blank reply. csvgeocode tries to look for specific fields in a reply that doesn't contain any data, and that's where this error message is coming from.

If I was competent at Javascript, I'd file a pull request to deal with a blank reply, but I'm afraid that I'm not. :-/ Thanks for helping me figure out what was going wrong!
